graal-tools/src/main/scala/io/cloudstate/graaltools/AkkaActorRegisterFeature.scala

```diff
@@ -1,30 +1,25 @@
 package io.cloudstate.graaltools
 
 import com.oracle.svm.core.annotate.AutomaticFeature
-import org.graalvm.nativeimage.hosted.Feature.QueryReachabilityAccess
 import org.graalvm.nativeimage.hosted.{Feature, RuntimeReflection}
 
-import scala.collection.JavaConverters._
-import java.util.concurrent.ConcurrentHashMap
-
 @AutomaticFeature
 final class AkkaActorRegisterFeature extends Feature {
-  private[this] final val cache = {
-    val c = ConcurrentHashMap.newKeySet[String]
+  private[this] final val ignoreClass = Set(
     // FIXME the following are configured in akka-graal-config, remove this when migrating from it
-    c.add("akka.io.TcpConnection")
-    c.add("akka.io.TcpListener")
-    c.add("akka.stream.impl.fusing.ActorGraphInterpreter")
-    c
-  }
+    "akka.io.TcpConnection",
+    "akka.io.TcpListener",
+    "akka.stream.impl.fusing.ActorGraphInterpreter"
+  ).contains(_)
 
-  override final def duringAnalysis(access: Feature.DuringAnalysisAccess): Unit = {
+  override final def beforeAnalysis(access: Feature.BeforeAnalysisAccess): Unit = {
     val akkaActorClass =
       access.findClassByName(classOf[akka.actor.Actor].getName) // We do this to get compile-time safety of the classes, and allow graalvm to resolve their names
-    if (akkaActorClass != null && access.isReachable(akkaActorClass)) {
+
+    def register(subtype: Class[_]) =
       for {
-        subtype <- access.reachableSubtypes(akkaActorClass).iterator.asScala
-        if subtype != null && !subtype.isInterface && cache.add(subtype.getName)
+        subtype <- Option(subtype)
+        if !subtype.isInterface && ignoreClass(subtype.getName)
```

Comment:
Looks like this should be `!ignoreClass`?

```suggestion
        if !subtype.isInterface && !ignoreClass(subtype.getName)
```

Reply:
Uuuhhh.... how does it work with that mistake in place? 🤯

Reply:
@dwijnand How are you testing that it works?

Reply:
`proxy-core/graalvm-native-image:packageBin; tck/it:test` together with

```diff
--- tck/src/it/resources/application.conf
+++ tck/src/it/resources/application.conf
@@ -1,4 +1,4 @@
-cloudstate-tck.verify = ["Akka + Node.js", "Akka + Java", "Akka + Scala", "Akka + Go", "Akka + Kotlin"]
+cloudstate-tck.verify = ["Akka + Java"]
 cloudstate-tck.combinations = [{
   name = "Akka + Node.js"
   tck {
@@ -44,8 +44,8 @@ cloudstate-tck.combinations = [{
     hostname = "127.0.0.1"
     port = 9000
     directory = ${user.dir}
-    command = ["java","-Xmx512M", "-Xms128M", "-Dconfig.resource=in-memory.conf", "-Dcloudstate.proxy.dev-mode-enabled=true", "-jar", "proxy/core/target/scala-2.12/akka-proxy.jar"]
-    //command = ["proxy/core/target/graalvm-native-image/cloudstate-proxy-core", "-Djava.library.path="${JAVA_HOME}"/lib", "-Dconfig.resource=in-memory.conf", "-Dcloudstate.proxy.dev-mode-enabled=true"]
+    //command = ["java","-Xmx512M", "-Xms128M", "-Dconfig.resource=in-memory.conf", "-Dcloudstate.proxy.dev-mode-enabled=true", "-jar", "proxy/core/target/scala-2.12/akka-proxy.jar"]
+    command = ["proxy/core/target/graalvm-native-image/cloudstate-proxy-core", "-Djava.library.path="${JAVA_HOME}"/lib", "-Dconfig.resource=in-memory.conf", "-Dcloudstate.proxy.dev-mode-enabled=true"]

     env-vars {
       USER_FUNCTION_PORT = "8090"
```

Reply:
That should work... What happens if you omit "-Dcloudstate.proxy.dev-mode-enabled=true"?

Reply:
It fails when running with the docker build:

```
> dockerBuildNativeDevMode publishLocal
...
Error: A field that was already processed by the analysis cannot be re-registered as writable: private final akka.actor.ActorContext akka.io.TcpConnection.context
```

Reply:
Same if I use a local build:

```
❯ set `proxy-core`/graalVMVersion := None
❯ proxy-core/graalvm-native-image:packageBin
...
[error] Error: A field that was already processed by the analysis cannot be re-registered as writable: private final akka.actor.ActorContext akka.io.TcpListener.context
```